pandas/core/ops.py

```diff
@@ -703,6 +702,23 @@ def wrapper(left, right, name=name, na_op=na_op):
     return wrapper
 
 
+def _dispatch_to_index_op(op, left, right, index_class):
+    """
+    Defer to DatetimeIndex implementations for type
+    checking and timezone handling.
+    """
+    left_idx = index_class(left)
+    left_idx.freq = None  # avoid accidentally allowing integer add/sub
```

Comment:
this is a mutating operation, don't do it

Reply:
This is necessary.  If I change this to an `assert left_idx.freq is None` that assertion fails in a number of tests:

```
dti = pd.date_range('1999-09-30', periods=10, tz='US/Pacific')
ser = pd.Series(dti)
>>> pd.DatetimeIndex(ser).freq
<Day>
```

It looks like this only occurs for tzaware cases.

Reply:
> It looks like this only occurs for tzaware cases.

Possibly related to caching of DatetimeIndexes and the fact that Series[datetime64tz]._data.blocks[0].values is itself a DatetimeIndex?